hedera-node/hedera-app/src/xtest/java/contract/MiscViewsXTest.java

```diff
@@ -0,0 +1,233 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package contract;
+
+import static com.hedera.node.app.service.contract.impl.ContractServiceImpl.CONTRACT_SERVICE;
+import static contract.AssortedOpsXTestConstants.ONE_HBAR;
+import static contract.MiscViewsXTestConstants.COINBASE_ID;
+import static contract.MiscViewsXTestConstants.ERC20_TOKEN_ID;
+import static contract.MiscViewsXTestConstants.ERC721_TOKEN_ID;
+import static contract.MiscViewsXTestConstants.ERC_USER_ADDRESS;
+import static contract.MiscViewsXTestConstants.ERC_USER_ID;
+import static contract.MiscViewsXTestConstants.GET_PRNG_SEED;
+import static contract.MiscViewsXTestConstants.NEXT_ENTITY_NUM;
+import static contract.MiscViewsXTestConstants.OPERATOR_ID;
+import static contract.MiscViewsXTestConstants.PRNG_SEED;
+import static contract.MiscViewsXTestConstants.SECRET;
+import static contract.MiscViewsXTestConstants.SPECIAL_QUERIES_X_TEST;
+import static contract.MiscViewsXTestConstants.VIEWS_INITCODE_FILE_ID;
+
+import com.esaulpaugh.headlong.abi.TupleType;
+import com.hedera.hapi.node.base.AccountID;
+import com.hedera.hapi.node.base.FileID;
+import com.hedera.hapi.node.base.NftID;
+import com.hedera.hapi.node.base.TokenID;
+import com.hedera.hapi.node.base.TokenType;
+import com.hedera.hapi.node.base.TransactionID;
+import com.hedera.hapi.node.contract.ContractCreateTransactionBody;
+import com.hedera.hapi.node.state.blockrecords.RunningHashes;
+import com.hedera.hapi.node.state.common.EntityIDPair;
+import com.hedera.hapi.node.state.common.EntityNumber;
+import com.hedera.hapi.node.state.contract.Bytecode;
+import com.hedera.hapi.node.state.contract.SlotKey;
+import com.hedera.hapi.node.state.contract.SlotValue;
+import com.hedera.hapi.node.state.file.File;
+import com.hedera.hapi.node.state.token.Account;
+import com.hedera.hapi.node.state.token.AccountApprovalForAllAllowance;
+import com.hedera.hapi.node.state.token.Nft;
+import com.hedera.hapi.node.state.token.Token;
+import com.hedera.hapi.node.state.token.TokenRelation;
+import com.hedera.hapi.node.transaction.TransactionBody;
+import com.hedera.node.app.spi.state.ReadableKVState;
+import com.hedera.pbj.runtime.io.buffer.Bytes;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.jetbrains.annotations.NotNull;
+
+public class MiscViewsXTest extends AbstractContractXTest {
+
+    @Override
+    protected void handleAndCommitScenarioTransactions() {
+        handleAndCommit(CONTRACT_SERVICE.handlers().contractCreateHandler(), synthCreateTxn());
+        var context =
+                handleAndCommitSingleTransaction(CONTRACT_SERVICE.handlers().contractCallHandler(), synthCallPrng());
+    }
```

Comment:
See my comment above on asserting bases on data in the record builders.  I noticed however that the tests do fail if an error is thrown which led me to find a couple of problems in the implementation for the system contract

Reply:
Still it would be nice to have an explicit assert on contract call success, revert, or halt, if it could be arranged (as part of `scenarioPasses` I guess, or maybe `handleAndCommit`.  (They'd have to take a parameter indicting the expected result for a particular call.)

Reply:
I think the best I can do for now is to assert that the recordbuilder object has a status of SUCCESS

Reply:
Because otherwise it is buried inside of `handle`?

Reply:
yes

Reply:
Luke can you keep a list of improvements to the xtest structure that we're talking about? This is one (being able to assert on the contract result success/revert/halt) - another is the one about running multiple separate _tests_ in a _scenario_ and keeping their assertions separate.